command/ca/provisioner/provisioner.go

```diff
@@ -72,3 +88,304 @@ $ step ca provisioner remove max@smallstep.com --kid 1234 --ca-config ca.json
 '''`,
 	}
 }
+
+type crudClient interface {
+	CreateProvisioner(prov *linkedca.Provisioner) (*linkedca.Provisioner, error)
+	GetProvisioner(opts ...ca.ProvisionerOption) (*linkedca.Provisioner, error)
+	UpdateProvisioner(name string, prov *linkedca.Provisioner) error
+	RemoveProvisioner(opts ...ca.ProvisionerOption) error
+}
+
+func newCRUDClient(cliCtx *cli.Context, cfgFile string) (crudClient, error) {
+	_, err := os.Stat(cfgFile)
+	switch {
+	case errors.Is(err, os.ErrNotExist):
+		return cautils.NewAdminClient(cliCtx)
+	case err == nil:
+		cfg, err := config.LoadConfiguration(cfgFile)
+		if err != nil {
+			return nil, fmt.Errorf("error loading configuration: %w", err)
+		}
+		if cfg.AuthorityConfig.EnableAdmin {
+			if len(cfg.AuthorityConfig.Provisioners) > 0 {
+				return nil, errors.New("when 'enableAdmin' attribute set to 'true', provisioners list in ca.json must be empty")
+			}
+			return cautils.NewAdminClient(cliCtx)
+		}
+		return newAdminAPIDisabledClient(context.Background(), cfg, cfgFile)
+	default:
+		return nil, errs.FileError(err, cfgFile)
+	}
+}
+
+func parseInstanceAge(ctx *cli.Context) (age string, err error) {
+	if !ctx.IsSet("instance-age") {
+		return
+	}
+	age = ctx.String("instance-age")
+	dur, err := time.ParseDuration(age)
+	if err != nil {
+		return "", err
+	}
+	if dur < 0 {
+		return "", errs.MinSizeFlag(ctx, "instance-age", "0s")
+	}
+	return
+}
+
+func removeElements(list, rems []string) []string {
+	if len(list) == 0 {
+		return list
+	}
+	for _, rem := range rems {
+		for i, elem := range list {
+			if elem == rem {
+				list[i] = list[len(list)-1]
+				list = list[:len(list)-1]
+				break
+			}
+		}
+	}
+	return list
+}
```

Comment:
This method doesn't remove duplicated elements in `list`. Is this something that can happen?

Reply:
I suppose it can now, because a user could manually add multiple elements to the config. And I'm not seeing any de-duping on `add`.

Reply:
It might be confusing, but I guess that was the previous behavior.